impl.go

```diff
@@ -170,8 +229,26 @@ func typeSpec(path string, id string, srcDir string) (Pkg, Spec, error) {
 				if spec.Name.Name != id {
 					continue
 				}
+				tParams := make(map[string]string, len(typeParams))
```

Comment:
This addition feels like it should perhaps be a helper function, with useful docs on it.